faker/providers/internet/__init__.py

```diff
@@ -169,15 +201,90 @@ def url(self):
         pattern = self.random_element(self.url_formats)
         return self.generator.parse(pattern)
 
-    def ipv4(self, network=False):
-        """Produce a random IPv4 address or network with a valid CIDR"""
+    def ipv4(self, network=False, address_class=None, private=None):
+        """
+        Produce a random IPv4 address or network with a valid CIDR.
+
+        :param network: Network address
+        :param address_class: IPv4 address class (a, b, or c)
+        :param private: Public or private
+        :returns: IPv4
+        """
+        if private is True:
+            return self.ipv4_private(address_class=address_class,
+                                     network=network)
+        elif private is False:
+            return self.ipv4_public(address_class=address_class,
+                                    network=network)
         address = str(ip_address(self.generator.random.randint(
```

Comment:
can we add the option to return any IP (without checking if public or private) from a specific address_class?

Reply:
I prefer to not provider address class for `ipv4()`, because I would like this method to provide any IP.
From private/public, from reserved blocks, etc

Reply:
the `address_class` will optional, and default to the old algorithm of returning any IP. We would filter by address class only if `address_class` is specified